internal/pkg/override/patch.go

```diff
@@ -0,0 +1,253 @@
+			for i := 0; i < len(node.Content); i += 2 {
+				if node.Content[i].Value == pointer[0] {
+					return y.encodeAndStop(node.Content[i+1])
+				}
+			}
+
+			// if the key isn't in this map, then we need to create it for them
+			node.Content = append(node.Content, &yaml.Node{
+				Kind:  yaml.ScalarNode,
+				Tag:   "!!str",
+				Value: pointer[0],
+			})
+			node.Content = append(node.Content, &y.Value)
+			return errStopFollowingPointer
+		case yaml.SequenceNode:
+			if pointer[0] == "-" || pointer[0] == "" {
+				// add to end of sequence
+				node.Content = append(node.Content, &y.Value)
+				return errStopFollowingPointer
+			}
+
+			idx, err := strconv.Atoi(pointer[0])
+			switch {
+			case err != nil:
+				return fmt.Errorf("expected index in sequence, got %q", pointer[0])
+			case idx < 0 || idx > len(node.Content)-1:
+				return fmt.Errorf("invalid index %d for sequence of length %d", idx, len(node.Content))
+			}
+
+			// add node at idx
+			node.Content = append(node.Content[:idx], append([]*yaml.Node{&y.Value}, node.Content[idx:]...)...)
+			return errStopFollowingPointer
+		}
+
+		return nil
+	})
+}
+
+func (y *yamlPatch) doRemove(root *yaml.Node) error {
+	return followJSONPointer(root, y.Path, func(node *yaml.Node, pointer []string) error {
+		if len(pointer) != 1 {
+			return nil
+		}
+
+		switch node.Kind {
+		case yaml.MappingNode:
+			for i := 0; i < len(node.Content); i += 2 {
+				if node.Content[i].Value == pointer[0] {
+					node.Content = append(node.Content[:i], node.Content[i+2:]...)
+					return errStopFollowingPointer
+				}
+			}
+		case yaml.SequenceNode:
+			idx, err := strconv.Atoi(pointer[0])
+			switch {
+			case err != nil:
+				return fmt.Errorf("expected index in sequence, got %q", pointer[0])
+			case idx < 0 || idx > len(node.Content)-1:
+				return fmt.Errorf("invalid index %d for sequence of length %d", idx, len(node.Content))
+			}
+
+			node.Content = append(node.Content[:idx], node.Content[idx+1:]...)
+			return errStopFollowingPointer
+		}
+
+		return nil
+	})
+}
+
+func (y *yamlPatch) applyReplace(root *yaml.Node) error {
+	return followJSONPointer(root, y.Path, func(node *yaml.Node, pointer []string) error {
+		if len(pointer) > 0 {
+			return nil
+		}
+		return y.encodeAndStop(node)
+	})
+}
+
+func (y *yamlPatch) encodeAndStop(node *yaml.Node) error {
+	if err := node.Encode(y.Value); err != nil {
+		return err
+	}
+	return errStopFollowingPointer
+}
+
+var errStopFollowingPointer = errors.New("stop following pointer")
+
+func followJSONPointer(root *yaml.Node, pointer string, visit func(node *yaml.Node, pointer []string) error) error {
+	split := strings.Split(pointer, "/")
+	for i := range split {
+		split[i] = strings.ReplaceAll(split[i], "~0", "~")
+		split[i] = strings.ReplaceAll(split[i], "~1", "/")
+	}
+
+	return followJSONPointerHelper(root, split, visit)
+}
+
+func followJSONPointerHelper(node *yaml.Node, pointer []string, visit func(node *yaml.Node, pointer []string) error) error {
```

Comment:
I wonder if the `visit` function should take as input the subpath traveled so far that in our errors we can provide exactly where the issue occurs?
```suggestion
func followJSONPointerHelper(node *yaml.Node, pointer []string, visit func(node *yaml.Node, traversed, remaining []string) error) error {
```
so that we can do something like:
```go
if err := visit(node, pointer); err != nil && !errors.Is(err, errSkip) {
   return fmt.Errorf("visit %q: %v", strings.Join(traversed, jsonPathSeparator), err)
}

// and further down
return fmt.Errorf("visit %q: key %q not found in map", strings.Join(traversed, jsonPathSeparator),pointer[0])
```